src/benchmark/operators/table_scan_sorted_benchmark.cpp

```diff
@@ -0,0 +1,285 @@
+                  {"String", std::make_pair(std::bind(sorted_string_table, std::placeholders::_1, std::placeholders::_2,
+                                                      std::placeholders::_3, string_size),
+                                            std::bind(unsorted_string_table, std::placeholders::_1,
+                                                      std::placeholders::_2, std::placeholders::_3, string_size))}};
+
+  const std::map<std::string, OrderByMode> order_bys{
+      {"AscendingNullsFirst", OrderByMode::Ascending},
+      {"AscendingNullsLast", OrderByMode::AscendingNullsLast},
+      {"DescendingNullsFirst", OrderByMode::Descending},
+      {"DescendingNullsLast", OrderByMode::DescendingNullsLast},
+  };
+
+  const std::map<std::string, EncodingType> encoding_types{{"None", EncodingType::Unencoded},
+                                                           {"Dictionary", EncodingType::Dictionary}};
+
+  for (const auto& table_type : table_types) {
+    const auto data_type = table_type.first;
+    const auto sorted_table = table_type.second.first;
+    const auto unsorted_table = table_type.second.second;
+
+    for (const auto& predicate : predicates) {
+      const auto predicate_name = predicate.first;
+      const auto predicate_condition = predicate.second;
+
+      for (const auto& order_by : order_bys) {
+        const auto order_by_name = order_by.first;
+        const auto order_by_mode = order_by.second;
+
+        for (const auto& encoding : encoding_types) {
+          const auto encoding_name = encoding.first;
+          const auto encoding_type = encoding.second;
+
+          for (const auto selectivity : selectivities) {
+            benchmark::RegisterBenchmark(
+                ("BM_TableScanSorted/" + data_type + "/" + predicate_name + "/" + order_by_name + "/" + encoding_name +
+                 "/" + std::to_string(selectivity) + "/" + "Sorted")
+                    .c_str(),
+                BM_TableScanSorted, rows, selectivity, predicate_condition, encoding_type, order_by_mode, sorted_table);
+            benchmark::RegisterBenchmark(
+                ("BM_TableScanSorted/" + data_type + "/" + predicate_name + "/" + order_by_name + "/" + encoding_name +
+                 "/" + std::to_string(selectivity) + "/" + "Unsorted")
+                    .c_str(),
+                BM_TableScanSorted, rows, selectivity, predicate_condition, encoding_type, std::nullopt,
+                unsorted_table);
+          }
+        }
+      }
+    }
+  }
+}
+
+// We need to call the registerTableScanSortedBenchmarks() to register the benchmarks. We could call it inside the
+// micro_benchmark_main.cpp::main() method, but then these benchmarks would also be included when building the
+// hyriseBenchmarkPlayground. Instead, we create a global object whose sole purpose is to register the benchmarks in its
+// constructor.
+class StartUp {
```

Comment:
Should this habben in the namespace of opossum?

Reply:
No :)